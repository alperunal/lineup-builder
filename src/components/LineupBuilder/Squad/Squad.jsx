import React from 'react';
import cn from 'classnames';
import { FormattedMessage } from 'react-intl';
import PropTypes from 'prop-types';

import GoalIcon from 'assets/icons/goal_icon.svg';
import YellowIcon from 'assets/icons/yellow_card_icon.svg';
import RedIcon from 'assets/icons/red_card_icon.svg';

import './Squad.scss';

const Squad = ({ players, editPlayer }) => (
  <div className="squad">
    <table className="player-table">
      <thead>
        <tr>
          <td className="player-table__name">
            <FormattedMessage id="squad.player" defaultMessage="Player" />
          </td>
          <td className="player-table__no">
            <FormattedMessage id="squad.no" defaultMessage="No" />
          </td>
          <td className="player-table__icon">
            <img src={GoalIcon} alt="Goal" />
          </td>
          <td className="player-table__icon">
            <img src={YellowIcon} alt="Yellow Card" />
          </td>
          <td className="player-table__icon">
            <img src={RedIcon} alt="Red Card" />
          </td>
        </tr>
      </thead>
      <tbody>
        {players.map((player) => (
          <tr key={player.id}>
            <td className={cn('player-table__name')}>
              <span className="form-group">
                <input
                  type="text"
                  spellCheck={false}
                  value={player.name}
                  onChange={(event) => editPlayer(
                    player.id,
                    event.target.value,
                    player.num,
                    player.position,
                    player.goal || false,
                    player.yellowCard || false,
                    player.redCard || false,
                  )}
                />
              </span>
            </td>
            <td className={cn('player-table__no')}>
              <span className="form-group">
                <input
                  type="text"
                  value={player.num}
                  maxLength={2}
                  onChange={(event) => editPlayer(
                    player.id,
                    player.name,
                    event.target.value,
                    player.position,
                    player.goal || false,
                    player.yellowCard || false,
                    player.redCard || false,
                  )}
                />
              </span>
            </td>
            <td className={cn('player-table__icon')}>
              <span className="form-group">
                <input
                  type="checkbox"
                  checked={player.goal}
                  onChange={(event) => editPlayer(
                    player.id,
                    player.name,
                    player.num,
                    player.position,
                    event.target.checked,
                    player.yellowCard || false,
                    player.redCard || false,
                  )}
                />
              </span>
            </td>
            <td className={cn('player-table__icon')}>
              <span className="form-group">
                <input
                  type="checkbox"
                  checked={player.yellowCard}
                  onChange={(event) => editPlayer(
                    player.id,
                    player.name,
                    player.num,
                    player.position,
                    player.goal || false,
                    event.target.checked,
                    player.redCard || false,
                  )}
                />
              </span>
            </td>
            <td className={cn('player-table__icon')}>
              <span className="form-group">
                <input
                  type="checkbox"
                  checked={player.redCard}
                  onChange={(event) => editPlayer(
                    player.id,
                    player.name,
                    player.num,
                    player.position,
                    player.goal || false,
                    player.yellowCard || false,
                    event.target.checked,
                  )}
                />
              </span>
            </td>
          </tr>
        ))}
      </tbody>
    </table>
  </div>
);

Squad.propTypes = {
  players: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      name: PropTypes.string.isRequired,
      position: PropTypes.shape({
        x: PropTypes.number.isRequired,
        y: PropTypes.number.isRequired,
      }).isRequired,
      goal: PropTypes.bool.isRequired,
      yellowCard: PropTypes.bool.isRequired,
      redCard: PropTypes.bool.isRequired,
    }),
  ).isRequired,
  editPlayer: PropTypes.func.isRequired,
};

export default Squad;
