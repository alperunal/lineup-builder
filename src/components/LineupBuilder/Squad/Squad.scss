@import 'styles/_variables.scss';

.squad {
  box-sizing: border-box;
  margin-top: 25px;

  .player-table {
    overflow-x: auto;

    &__name {
      width: 30%;
      input {
        width: 100%;
      }
    }

    &__no {
      width: 15%;
      input {
        width: 100%;
      }
    }

    &__icon {
      width: 10%;
      padding-left: 5px;
      text-align: center;

      input {
        display: block;
        margin: auto;
      }
    }

    img {
      height: 24px;
    }
  }
}
