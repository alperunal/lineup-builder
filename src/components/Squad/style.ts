import styled from 'styled-components';

export const Select = styled.select`
    width: 100%;
    padding-left: 10px;
    height: 40px;
    background: rgba(0, 0, 0, 0.5);
    color: #fff;
    border: 4px solid rgba(0, 0, 0, 0.1);
    border-radius: 5px;
    font-family: 'Roboto';
    font-size: 20px;
`;

export const Input = styled.input`
    /* width: 300px; */
    padding-left: 10px;
    height: 40px;
    background: rgba(0, 0, 0, 0.5);
    color: #fff;
    border: 4px solid rgba(0, 0, 0, 0.1);
    border-radius: 5px;
    font-family: 'Roboto';
    font-size: 20px;
    box-sizing: border-box;
    margin-bottom: 5px;

    &::placeholder {
        color: #ddd;
    }
`;

export const PositionList = styled.ul`
    list-style-type: none;
    margin: 20px 0 0 0;
    padding: 0;
`;

export const PositionListItem = styled.li`
    display: flex;
`;
