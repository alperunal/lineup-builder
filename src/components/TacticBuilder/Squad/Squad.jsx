import React from 'react';
import cn from 'classnames';
import { FormattedMessage } from 'react-intl';

import GoalIcon from 'assets/icons/goal_icon.svg';
import YellowIcon from 'assets/icons/yellow_card_icon.svg';
import RedIcon from 'assets/icons/red_card_icon.svg';

import './Squad.module.scss';

const Squad = ({ players, editPlayer }) => (
    <div className="squad">
        <table className="player-table">
            <thead>
                <tr>
                    <td className="player-table__name">
                        <FormattedMessage
                            id="squad.player"
                            defaultMessage="Player"
                        />
                    </td>
                    <td className="player-table__no">
                        <FormattedMessage
                            id="squad.no"
                            defaultMessage="No"
                        />
                    </td>
                    <td className="player-table__icon">
                        <img src={GoalIcon} alt="Goal" />
                    </td>
                    <td className="player-table__icon">
                        <img src={YellowIcon} alt="Yellow Card" />
                    </td>
                    <td className="player-table__icon">
                        <img src={RedIcon} alt="Red Card" />
                    </td>
                </tr>
            </thead>
            <tbody>
                {players.map((player) => (
                    <tr key={player.id}>
                        <td
                            className={
                                cn(
                                    'player-table__name',
                                )
                            }
                        >
                            <span className="form-group">
                                <input
                                    type="text"
                                    spellCheck={false}
                                    value={player.name}
                                    onChange={(event) => editPlayer(player.id, event.target.value, player.num, player.position, player.goal || false, player.yellowCard || false, player.redCard || false)}
                                />
                            </span>
                        </td>
                        <td
                            className={
                                cn(
                                    'player-table__no',
                                )
                            }
                        >
                            <span className="form-group">
                                <input
                                    type="text"
                                    value={player.num}
                                    maxLength={2}
                                    onChange={(event) => editPlayer(player.id, player.name, event.target.value, player.position, player.goal || false, player.yellowCard || false, player.redCard || false)}
                                />
                            </span>
                        </td>
                        <td
                            className={
                                cn(
                                    'player-table__icon',
                                )
                            }
                        >
                            <span className="form-group">
                                <input
                                    type="checkbox"
                                    checked={player.goal}
                                    onChange={(event) => editPlayer(player.id, player.name, player.num, player.position, event.target.checked, player.yellowCard || false, player.redCard || false)}
                                />
                            </span>
                        </td>
                        <td
                            className={
                                cn(
                                    'player-table__icon',
                                )
                            }
                        >
                            <span className="form-group">
                                <input
                                    type="checkbox"
                                    checked={player.yellowCard}
                                    onChange={(event) => editPlayer(player.id, player.name, player.num, player.position, player.goal || false, event.target.checked, player.redCard || false)}
                                />
                            </span>
                        </td>
                        <td
                            className={
                                cn(
                                    'player-table__icon',
                                )
                            }
                        >
                            <span className="form-group">
                                <input
                                    type="checkbox"
                                    checked={player.redCard}
                                    onChange={(event) => editPlayer(player.id, player.name, player.num, player.position, player.goal || false, player.yellowCard || false, event.target.checked)}
                                />
                            </span>
                        </td>
                    </tr>
                ))}
            </tbody>
        </table>
    </div>
);

export default Squad;
