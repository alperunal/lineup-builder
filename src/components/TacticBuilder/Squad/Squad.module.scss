.squad {
    box-sizing: border-box;

    .player-table {
        overflow-x: auto;

        &__name {
            width: 200px;
        }

        &__no {
            width: 60px;
        }

        &__icon {
            padding-left: 5px;

            @media (min-width: 1000px) {
                padding-left: 15px;
            }
        }
    }
}
