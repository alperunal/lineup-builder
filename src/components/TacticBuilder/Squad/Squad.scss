.squad {
    box-sizing: border-box;
    padding: 10px;
    max-width: 400px;

    &__position-list {
        list-style-type: none;
        margin: 20px 0 0 0;
        padding: 0;
    }

    &__position-list-item {
        display: flex;
    }
}