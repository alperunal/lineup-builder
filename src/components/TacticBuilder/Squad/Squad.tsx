import React from 'react';
import { IPlayer, IPosition } from 'constants/model';
import GoalIcon from 'assets/icons/goal_icon.svg';
import YellowIcon from 'assets/icons/yellow_card_icon.svg';
import RedIcon from 'assets/icons/red_card_icon.svg';
import './Squad.module.scss';

interface IProps {
    players: IPlayer[];
    editPlayer: (id: string, nName: string, nNum: string, nPosition: IPosition, nGoal: boolean, nYellowCard: boolean, nRedCard: boolean) => void;
}

const Squad: React.FC<IProps> = ({ players, editPlayer }: IProps) => (
    <div className="squad">
        <table className="player-table">
            <thead>
                <tr>
                    <td>Name</td>
                    <td># No</td>
                    <td><img src={GoalIcon} alt="Goal" /></td>
                    <td><img src={YellowIcon} alt="Yellow Card" /></td>
                    <td><img src={RedIcon} alt="Red Card" /></td>
                </tr>
            </thead>
            <tbody>
                {players.map((player: IPlayer) => (
                    <tr key={player.id}>
                        <td className="form-group">
                            <input
                                type="text"
                                spellCheck={false}
                                value={player.name}
                                onChange={(event: React.ChangeEvent<HTMLInputElement>) => editPlayer(player.id, event.target.value, player.num, player.position, player.goal || false, player.yellowCard || false, player.redCard || false)}
                            />
                        </td>
                        <td className="form-group">
                            <input
                                type="text"
                                value={player.num}
                                maxLength={2}
                                onChange={(event: React.ChangeEvent<HTMLInputElement>) => editPlayer(player.id, player.name, event.target.value, player.position, player.goal || false, player.yellowCard || false, player.redCard || false)}
                            />
                        </td>
                        <td className="form-group">
                            <input
                                type="checkbox"
                                onChange={(event: React.ChangeEvent<HTMLInputElement>) => editPlayer(player.id, player.name, player.num, player.position, event.target.checked, player.yellowCard || false, player.redCard || false)}
                            />
                        </td>
                        <td className="form-group">
                            <input
                                type="checkbox"
                                onChange={(event: React.ChangeEvent<HTMLInputElement>) => editPlayer(player.id, player.name, player.num, player.position, player.goal || false, event.target.checked, player.redCard || false)}
                            />
                        </td>
                        <td className="form-group">
                            <input
                                type="checkbox"
                                onChange={(event: React.ChangeEvent<HTMLInputElement>) => editPlayer(player.id, player.name, player.num, player.position, player.goal || false, player.yellowCard || false, event.target.checked)}
                            />
                        </td>
                    </tr>
                ))}
            </tbody>
        </table>
    </div>
);

export default Squad;
