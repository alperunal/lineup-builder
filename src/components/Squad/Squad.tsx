import React from 'react';
import { IPlayer, IPosition } from '../../constants/model';
import { Input } from '../../style/VoetUI';
import './Squad.scss';

interface IProps {
    players: IPlayer[];
    editPlayer: (id: string, nName: string, nNum: string, nPosition: IPosition) => void;
}

const Squad = ({ players, editPlayer }: IProps) => {
    return (
        <div className="squad">
            <div>
                <ul className="squad__position-list">
                    {players.map(player => (
                        <li className="squad__position-list-item" key={player.id}>
                            <Input
                                type="text"
                                style={{ width: '80%', marginRight: '5px' }}
                                spellCheck={false}
                                value={player.name}
                                onChange={(event: React.ChangeEvent<HTMLInputElement>) => editPlayer(player.id, event.target.value, player.num, player.position)}
                            />
                            <Input
                                type="text"
                                style={{ width: '20%' }}
                                value={player.num}
                                maxLength={2}
                                onChange={(event: React.ChangeEvent<HTMLInputElement>) => editPlayer(player.id, player.name, event.target.value, player.position)}
                            />
                        </li>
                    ))}
                </ul>
            </div>
        </div>
    );
};

export default Squad;
