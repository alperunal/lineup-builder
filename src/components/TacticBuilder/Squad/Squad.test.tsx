import React from 'react';
import { shallow } from 'enzyme';
import Squad from './Squad';

it('should render Squad correctly', () => {
    const props = {
        players: [],
        editPlayer: jest.fn(),
    };
    const wrapper = shallow(<Squad {...props} />);
    expect(wrapper).toMatchSnapshot();
});
