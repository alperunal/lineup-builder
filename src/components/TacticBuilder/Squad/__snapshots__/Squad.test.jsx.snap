// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Squad correctly 1`] = `ShallowWrapper {}`;
